programs/programs/policyengine/policy_engine.py

```diff
@@ -105,35 +115,44 @@ def pe_input(screen: Screen, programs: List[PolicyEngineCalulator]):
             if hasattr(program, "pe_output_period") and Data in program.pe_outputs:
                 period = program.pe_output_period
 
-            if not Data.member:
-                data = Data(screen, members, relationship_map)
-                value = data.value()
-                unit = raw_input["household"][data.unit][data.sub_unit]
+            if issubclass(Data, Member):
+                for member in members:
+                    member_id = str(member.id)
+                    data = Data(screen, member, relationship_map)
+                    unit = raw_input["household"][data.unit][member_id]
+
+                    update_unit(unit, data, period)
+            elif issubclass(Data, TaxUnit):
+                # split the household into the main and secondary tax unit.
+                data = Data(screen, main_tax_members, relationship_map)
+                unit = raw_input["household"][data.unit][MAIN_TAX_UNIT]
 
-                if data.field in unit and period in unit[data.field]:
-                    if value != unit[data.field][period]:
-                        raise DependencyError(data.field, value, unit[data.field][period])
+                update_unit(unit, data, period)
 
-                if data.field not in unit:
-                    unit[data.field] = {}
+                data = Data(screen, secondary_tax_members, relationship_map)
+                unit = raw_input["household"][data.unit][SECONDARY_TAX_UNIT]
 
-                unit[data.field][period] = value
-                continue
+                update_unit(unit, data, period)
+            else:
+                data = Data(screen, members, relationship_map)
+                unit = raw_input["household"][data.unit][data.sub_unit]
 
-            for member in members:
-                member_id = str(member.id)
-                data = Data(screen, member, relationship_map)
-                value = data.value()
+                update_unit(unit, data, period)
 
-                unit = raw_input["household"][data.unit][member_id]
+    # delete the second tax unit if it is empty because PE can't handle empty tax units
+    if len(secondary_tax_members) == 0:
+        del raw_input["household"]["tax_units"][SECONDARY_TAX_UNIT]
 
-                if data.field in unit and period in unit[data.field]:
-                    if value != unit[data.field][period]:
-                        raise DependencyError(data.field, value, unit[data.field][period])
+    return raw_input
 
-                if data.field not in unit:
-                    unit[data.field] = {}
 
-                unit[data.field][period] = value
+def update_unit(unit, data: PolicyEngineCalulator, period: str):
```

Comment:
What does this function do? Is `period` a year or month?

Reply:
The period is the year.

This function adds the value of data calculated with the `value` method, and adds it to the `unit` in the raw data. e.x. that takes in a member unit, and adds the earned income. In this example `data.value = 100` and `data.field = "earned_income"`

```py
update_unit({"age": {"2024": 44}}, data, "2024")

# will make the unit:
{"age": {"2024": 44}, "earned_income": {"2024": 100}}
```